*** ./mpv-livetweet.lua	2015-02-17 21:52:09.000000000 +0100
--- ./mpv-livetweet-osx.lua	2015-02-17 21:50:37.000000000 +0100
*************** function tweet()
*** 34,37 ****
--- 34,63 ----
  	mp.osd_message("Screenshot tweeted!")
  end
  
+ function tweetwithtext()
+ 	local shot = os.tmpname()
+ 	os.remove(shot)
+ 	mp.commandv("screenshot_to_file", shot, "subtitles")
+ 	local openshot = io.open(shot)
+ 	local img_data = openshot:read("*a")
+ 	openshot:close()
+ 	os.remove(shot)
+ 
+ 	local text = io.popen('/usr/bin/osascript -e \'tell application "mpv" to set tweet to text returned of (display dialog "" with title "Tweet body" default answer "" buttons "Tweet" default button 1)\' -e \'do shell script "echo " & quoted form of tweet\'')
+ 	local body = text:read("*a")
+ 	text:close()
+ 
+ 	mp.resume()
+ 
+ 	local client = twitter.new(keys)
+ 	local tw, headers = client:tweet_with_media{
+ 		status = body, ["media[]"] = {
+ 			filename = shot,
+ 			data = img_data,
+ 		},
+ 	}
+ 	mp.osd_message("Screenshot tweeted!")
+ end
+ 
  mp.add_key_binding("alt+w", "tweet", function() tweet() end)
+ mp.add_key_binding("shift+alt+w", "tweet_with_text", function() tweetwithtext() end)
